### PR TITLE
Add Tagging to sources

### PR DESCRIPTION
**Pull request by agrare**
Add tags and tagging intermediate tables to sources

Depends on: https://github.com/RedHatInsights/insights-api-common-rails/pull/158

Relates to JIRA: TPINVTRY-815

### CODE REVIEW

File: app/models/authentication_tag.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class AuthenticationTag < ApplicationTag
```
Comment (bdunne) [inline app/models/authentication_tag.rb]: This should be `< ApplicationRecord`

File: app/models/endpoint_tag.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class EndpointTag < ApplicationTag
```
Comment (bdunne) [inline app/models/endpoint_tag.rb]: Same here

File: app/models/source_tag.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class SourceTag < ApplicationTag
```
Comment (bdunne) [inline app/models/source_tag.rb]: And here

File: app/models/concerns/act_as_taggable_on.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
+module ActAsTaggableOn
```
Comment (bdunne) [inline app/models/concerns/act_as_taggable_on.rb]: What do you think of moving this into InsightsApiCommon?

**Comment by bdunne** [comment]
Any tests?

**Comment by agrare** [comment]
Nope not yet

**Pull request closed without merging**